src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java

```diff
@@ -52,4 +62,18 @@ public void readsSinkTest(String inputBam, String outputFile) throws IOException
         Assert.assertEquals(rddParallelReads.count(), rddParallelReads2.count());
     }
 
-}
\ No newline at end of file
+    @Test(dataProvider = "loadReadsADAM", groups = "spark")
+    public void readsSinkADAMTest(String inputBam, String outputFile) throws IOException {
+        new File(outputFile).deleteOnExit();
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+
+        ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        JavaRDD<GATKRead> rddParallelReads = readSource.getParallelReads(inputBam);
+        SAMFileHeader header = ReadsSparkSource.getHeader(ctx, inputBam);
+
+        ReadsSparkSink.writeReads(ctx, outputFile, rddParallelReads, header, ReadsWriteFormat.ADAM);
+
+        JavaRDD<GATKRead> rddParallelReads2 = readSource.getADAMReads(outputFile, header);
+        Assert.assertEquals(rddParallelReads.count(), rddParallelReads2.count());
```

Comment:
I think you can actually compare the elements (not just the sizes). The reason why I didn't do that above is that a SAMRecord-backed GATKRead isn't the same as a Google model-backed GATKRead.